internal/pkg/cli/pipeline_override.go

```diff
@@ -0,0 +1,155 @@
+	"github.com/aws/copilot-cli/internal/pkg/aws/sessions"
+	"github.com/aws/copilot-cli/internal/pkg/config"
+	"github.com/aws/copilot-cli/internal/pkg/term/log"
+	termprogress "github.com/aws/copilot-cli/internal/pkg/term/progress"
+	"github.com/aws/copilot-cli/internal/pkg/term/prompt"
+	"github.com/aws/copilot-cli/internal/pkg/term/selector"
+	"github.com/aws/copilot-cli/internal/pkg/workspace"
+	"github.com/spf13/afero"
+	"github.com/spf13/cobra"
+)
+
+type overridePipelineOpts struct {
+	*overrideOpts
+
+	// Interfaces to interact with dependencies.
+	ws       wsPipelineReader
+	wsPrompt wsPipelineSelector
+}
+
+func newOverridePipelineOpts(vars overrideVars) (*overridePipelineOpts, error) {
+	fs := afero.NewOsFs()
+	ws, err := workspace.Use(fs)
+	if err != nil {
+		return nil, err
+	}
+
+	sessProvider := sessions.ImmutableProvider(sessions.UserAgentExtras("pipeline override"))
+	defaultSess, err := sessProvider.Default()
+	if err != nil {
+		return nil, fmt.Errorf("default session: %v", err)
+	}
+
+	prompt := prompt.New()
+
+	cmd := &overridePipelineOpts{
+		overrideOpts: &overrideOpts{
+			overrideVars: vars,
+			fs:           fs,
+			cfgStore:     config.NewSSMStore(identity.New(defaultSess), ssm.New(defaultSess), aws.StringValue(defaultSess.Config.Region)),
+			prompt:       prompt,
+			cfnPrompt:    selector.NewCFNSelector(prompt),
+			spinner:      termprogress.NewSpinner(log.DiagnosticWriter),
+		},
+		ws:       ws,
+		wsPrompt: selector.NewWsPipelineSelector(prompt, ws),
+	}
+	cmd.overrideOpts.packageCmd = cmd.newPipelinePackageCmd
+	return cmd, nil
+}
+
+// Validate returns an error for any invalid optional flags.
+func (o *overridePipelineOpts) Validate() error {
+	if err := o.overrideOpts.Validate(); err != nil {
+		return err
+	}
+	return o.validatePipelineName()
+}
+
+// Ask prompts for and validates any required flags.
+func (o *overridePipelineOpts) Ask() error {
+	if o.name == "" {
+		if err := o.askPipelineName(); err != nil {
+			return err
+		}
+	}
+	return o.overrideOpts.Ask()
+}
+
+// Execute writes IaC override files to the local workspace.
+func (o *overridePipelineOpts) Execute() error {
+	o.overrideOpts.dir = func() string {
+		return o.ws.PipelineOverridesPath(o.name)
+	}
+	return o.overrideOpts.Execute()
+
+}
+
+func (o *overridePipelineOpts) validatePipelineName() error {
+	if o.name == "" {
+		return nil
+	}
+	pipelines, err := o.ws.ListPipelines()
+	if err != nil {
+		return fmt.Errorf("list pipelines in the workspace: %v", err)
+	}
+	var names []string
+	for _, pipeline := range pipelines {
+		names = append(names, pipeline.Name)
+	}
+	if !contains(o.name, names) {
+		return fmt.Errorf("pipeline %q does not exist in the workspace", o.name)
+	}
+	return nil
+}
+
+func (o *overridePipelineOpts) askPipelineName() error {
+	pipeline, err := o.wsPrompt.WsPipeline("Which pipeline's resources would you like to override?", "")
+	fmt.Printf("The pipeline which you want to override is %v", pipeline.Name)
```

Comment:
i think this still needs to get removed 😛

Reply:
My bad, I will remove it rn🤦‍♂️